File: src/main/java/client/messages/quest/QuestRecordExMessage.java
package client.messages.quest;

import client.messages.MessageType;
import util.packet.PacketWriter;

public class QuestRecordExMessage extends AbstractQuestMessage {

    private final String value;

    public QuestRecordExMessage(short questId, String value) {
        super(questId);
        this.value = value;
    }

    @Override
    public MessageType getType() {
        return MessageType.QUEST_RECORD_EX_MESSAGE;
    }

    @Override
    protected void encodeData(PacketWriter pw) {
        super.encodeData(pw);
        pw.writeMapleString(value);
    }
}

File: src/main/java/client/messages/quest/QuestRecordExMessage.kt
package client.messages.quest

import client.messages.MessageType
import util.packet.PacketWriter

class QuestRecordExMessage(questId: Short, private val value: String) : AbstractQuestMessage(questId) {

    override val type: MessageType get() = MessageType.QUEST_RECORD_EX_MESSAGE

    override fun encodeData(pw: PacketWriter) {
        super.encodeData(pw)
        pw.writeMapleString(value)
    }
}